test/src/tests-kit-blocking.ts

```diff
@@ -616,6 +617,24 @@ describe('kit blocking', () => {
             })
         });
 
+        it('integration test - should not throw an error when unplanned user attributes are allowed and block.ua = true', function(done) {
+            window.mParticle.config.kitConfigs.push(forwarderDefaultConfiguration('MockForwarder'));
+            window.mParticle.init(apiKey, window.mParticle.config);
+            
+            //save old data points for reset later
+            const oldDataPoints = dataPlan.dtpn.vers.version_document.data_points;
+            dataPlan.dtpn.vers.version_document.data_points = [];
+            let kitBlocker = new KitBlocker(kitBlockerDataPlan, window.mParticle.getInstance());
+
+            expect(() => { kitBlocker.isAttributeKeyBlocked('unplannedAttr') }).to.not.throw(TypeError, /Cannot read properties of undefined \(reading 'unplannedAttr'\)/)
+            
```

Comment:
Add another expect for kitBlocker.isAttributeKeyBlocked('unplannedAttr' to equal true)

Reply:
added the case and confirmed this should be expected to be false